packages/twenty-server/src/modules/calendar/calendar-event-import-manager/jobs/calendar-relaunch-failed-calendar-channel.job.ts

```diff
@@ -0,0 +1,69 @@
+import { Scope } from '@nestjs/common';
+
+import { Process } from 'src/engine/core-modules/message-queue/decorators/process.decorator';
+import { Processor } from 'src/engine/core-modules/message-queue/decorators/processor.decorator';
+import { MessageQueue } from 'src/engine/core-modules/message-queue/message-queue.constants';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import {
+  CalendarChannelSyncStage,
+  CalendarChannelSyncStatus,
+  CalendarChannelWorkspaceEntity,
+} from 'src/modules/calendar/common/standard-objects/calendar-channel.workspace-entity';
+import { AccountsToReconnectService } from 'src/modules/connected-account/services/accounts-to-reconnect.service';
+
+export type CalendarRelaunchFailedCalendarChannelJobData = {
+  workspaceId: string;
+  calendarChannelId: string;
+};
+
+@Processor({
+  queueName: MessageQueue.calendarQueue,
+  scope: Scope.REQUEST,
+})
+export class CalendarRelaunchFailedCalendarChannelJob {
+  constructor(
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly accountsToReconnectService: AccountsToReconnectService,
+  ) {}
+
+  @Process(CalendarRelaunchFailedCalendarChannelJob.name)
+  async handle(data: CalendarRelaunchFailedCalendarChannelJobData) {
+    const { workspaceId, calendarChannelId } = data;
+
+    const calendarChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<CalendarChannelWorkspaceEntity>(
+        workspaceId,
+        'calendarChannel',
+        { shouldBypassPermissionChecks: true },
+      );
+
+    const calendarChannel = await calendarChannelRepository.findOne({
+      where: {
+        id: calendarChannelId,
+      },
+      relations: {
+        connectedAccount: {
+          accountOwner: true,
+        },
+      },
+    });
+
+    if (
+      !calendarChannel ||
+      calendarChannel.syncStage !== CalendarChannelSyncStage.FAILED
+    ) {
+      return;
+    }
+
+    await calendarChannelRepository.update(calendarChannelId, {
+      syncStage: CalendarChannelSyncStage.CALENDAR_EVENT_LIST_FETCH_PENDING,
+      syncStatus: CalendarChannelSyncStatus.ACTIVE,
+    });
+
+    await this.accountsToReconnectService.removeAccountToReconnect(
```

Comment:
@neo773 as we are handling FAILED status and we said that this banner should only be there for missing permissions, should we have this call here?